exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/ResultSetLoader.java

```diff
@@ -41,6 +42,12 @@
 
   public static final int DEFAULT_ROW_COUNT = BaseValueVector.INITIAL_VALUE_ALLOCATION;
 
+  /**
+   * Context for error messages.
+   */
+
+  RowSetContext context();
```

Comment:
Can this context be used for some this else rather than errors?
If not, maybe its better to name it `errorContext` everywhere?

Reply:
Renamed the class. Then, restructured the class to be more generic and moved it into common, alongside UserException. Doing so allowed simplifying usage of the context.